src/App.tsx

```diff
@@ -1,14 +1,79 @@
 /* eslint-disable max-len */
-import React from 'react';
+import {
+  FC,
+  useEffect,
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos, getUser } from './api';
+import { User } from './types/User';
+import { Filter } from './types/Filter';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [hasError, setHasError] = useState(false);
+  const [isModalShown, setIsModalShown] = useState(false);
+  const [filterBy, setFilterBy] = useState<Filter>(Filter.All);
+  const [query, setQuery] = useState<string>('');
+
+  const loadTodos = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos();
+
+      setTodos(todosFromServer);
+    } catch {
+      setHasError(true);
+    }
+  }, []);
+
+  const handleShowModal = (userId: number, todo: Todo) => {
+    setIsModalShown(true);
+    getUser(userId).then(userFromServer => {
```

Comment:
just choose one method & use just one of them
1. async / await 
or
2. then / catch